### PR TITLE
Redirect dead player LOOC to deadchat

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
The intent behind the combination of OOC being disabled but LOOC being enabled means we're concerned about players bitching or metagaming in the middle of the round. This redirects dead LOOCs to deadchat if OOC is disabled, while still letting living players talk in LOOC.

https://user-images.githubusercontent.com/60792108/180668951-585758c8-7ba3-4680-b0e9-a545b0980a31.mp4

**Changelog**

:cl: 
- add: If OOC is disabled in the middle of a round, dead players lose the ability to talk in LOOC.

### CODE REVIEW

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 1]
```diff
@@ -153,7 +155,13 @@ private void OnGameChange(GameRunLevelChangedEvent ev)
 
         message = SanitizeInGameOOCMessage(message);
 
-        switch (type)
+        var sendType = type;
+        // If we have OOC disabled for players, redirect dead/ghost LOOC to dead chat.
+        if (!_adminManager.IsAdmin(player) && !_configurationManager.GetCVar(CCVars.OocEnabled)
+            && (HasComp<GhostComponent>(source) || _mobStateSystem.IsDead(source)))
+            sendType = InGameOOCChatType.Dead;
```
Comment (mirrorcult) [inline Content.Server/Chat/Systems/ChatSystem.cs:162]: i feel like ghost LOOC should go to deadchat regardless of if OOC is disabled or not

**Comment by moonheart08** [comment]
this should itself be another CVar or set of CVars. For example on HRP LOOC is sometimes used for general conversation, even with ghosts. Maybe look into how to make this more configurable overall for how LOOC/OOC/dchat should behave.

**Comment by Elijahrane** [comment]
ok I made it a cvar

**Comment by Elijahrane** [comment]
> I did LOOC as observegang ghost to living... merely with intent to teach the game.

It's abused orders of magnitude more often than this ever happens. Someone can add mentor chat or just change the cvars per server

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**